src/services/file.rs

```diff
@@ -27,24 +27,18 @@ pub struct FileService;
 
 impl FileService {
     /// Get a file content from the remote git repository.
-    pub async fn get(ctx: Arc<Context>, id: Uuid, reference: String, path: String) -> Result<Content> {
+    pub async fn get(ctx: Arc<Context>, id: Uuid, path: String) -> Result<Content> {
         let playbook = ctx.client.playbooks().get(&id.to_string()).map_err(ApiError::NotFoundPlaybook)?;
         let source = playbook.preface.repository.unwrap();
 
         ctx.github_client
             .contents()
-            .find(&utils::repo(&source.repo)?, &path, &reference)
+            .find(&utils::repo(&source.repo)?, &path, &source.branch.unwrap_or_default())
```

Comment:
`&source.reference().unwrap()` You can determine if a reference is null before using it, and if it is null, just return an error.

Reply:
like is: 
```
let source = playbook.preface.repository;
if source.refernece().is_none() {
...
}
```